tests/integration/utils.sh

```diff
@@ -0,0 +1,132 @@
+declare -g SAMPLES_DIR   # has sample files used accross the integration tests
+declare -g KWROOT_DIR    # local kw dir to be copied to and installed in the containers
+declare -g DISTROS       # distributions we will run the integration tests
+
+# ensure path to directories is absolute
+script_dir=$(realpath "$(dirname "${0}")")
+CONTAINER_DIR="${script_dir}/podman"
+SAMPLES_DIR="${script_dir}/samples"
+KWROOT_DIR=$(realpath "${script_dir}/../..")
+
+# supported distros
+DISTROS=('archlinux' 'debian' 'fedora')
+
+# Builds a container imagem for the given distro
+function build_distro_image()
+{
+  local distro="${1}"
+  local file="${CONTAINER_DIR}/Containerfile_${distro}"
+
+  podman image build --file "$file" --tag "kw-${distro}" 2> /dev/null
+}
+
+# Builds container images and create containers used accross the tests.
+function setup_container_environment()
+{
+  local container_img
+  local container_name
+  local distro
+  local sha_container
+  local sha_kw
+  local working_directory
+  local output # output of podman used to check if containers exist or not
+
+  for distro in "${DISTROS[@]}"; do
+    # Only build the image if it does not exists. That's because trying to build
+    # the podman image takes a half second even if it exists and is cached.
+    if ! podman image exists "kw-${distro}"; then
+      # Build the image or exit. The integration tests cannot continue otherwise
+      if ! build_distro_image "$distro" > /dev/null; then
+        complain 'failed to setup container environment'
+        teardown_container_environment
+        exit 1
+      fi
+    fi
+
+    # the name of the container and the container image.
+    container_img="kw-${distro}"
+    container_name="kw-${distro}"
+
+    # Get running containers matching the container name.
+    # The output of the command will be a list of container names, one per line
+    output=$(podman container list --filter name="${container_name}" --format '{{.Names}}')
+
+    # If container is running, we do not recreate it for optimization purposes
+    # unless it is outdated.
+    if [[ "${output}" == "${container_name}" ]]; then
+      # to check if it is outdated, we check kw's version refers to the same
+      # commit as this local kw repo.
+
+      # first, get the sha inside the container
+      sha_container=$(podman exec "${container_name}" kw --version | grep Commit | sed 's/Commit: //')
+
+      # get the sha of this repo
+      sha_kw=$(git --git-dir "${KWROOT_DIR}/.git" rev-parse --short HEAD)
+
+      # if container is up to date, we will reuse it.
+      # Otherwise, destroy the container and create a new one.
+      if [[ "${sha_kw}" == "${sha_container}" ]]; then
+        continue
+      else
+        teardown_single_container "${container_name}"
+      fi
+    fi
+
+    # Maybe the container is not running, but it does exist.
+    # In this case, we tear it down because it could be in a improper state.
+    if podman container exists "${container_name}"; then
```

Comment:
Put this command outside the `if` condition, and just check `$?` (also use ``[[]]``).